src/platform_impl/macos/app_state.rs

```diff
@@ -79,36 +80,38 @@ where
 }
 
 #[derive(Default)]
-struct Handler {
+struct Handler<'a> {
     ready: AtomicBool,
     in_callback: AtomicBool,
     control_flow: Mutex<ControlFlow>,
     control_flow_prev: Mutex<ControlFlow>,
     start_time: Mutex<Option<Instant>>,
     callback: Mutex<Option<Box<dyn EventHandler>>>,
-    pending_events: Mutex<VecDeque<Event<Never>>>,
-    deferred_events: Mutex<VecDeque<Event<Never>>>,
+    pending_events: Mutex<VecDeque<Event<'a, Never>>>,
```

Comment:
rustc demanded a lifetime. `Event` in `pending_events` and `deferred_events` get `'static` through `lazy_static!` initialization of `Handler`. I guess, it's better to remove `'a` and set `Event` to have `'static` explicitly in order to not pollute code with lifetimes.

Reply:
Ah, OK. I'd switch to `'static` since having the lifetime in `Handler` states that the `Event`'s lifetime will last as long as the `Handler`, which for lifetime-d events isn't the case.